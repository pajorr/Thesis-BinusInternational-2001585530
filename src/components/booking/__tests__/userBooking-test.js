import React from 'react';
import { shallow } from 'enzyme';

import Booking from '../userBooking';


describe('userLogin', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<Booking debug/>);

        expect(component).toMatchSnapshot();
    });
});

//lifecycle methods

describe('userLogin', () => {
    it('calls componentDidMount', () => {
        jest.spyOn(Booking.prototype, 'componentDidMount');
        const wrapper = shallow(<Booking />);
        expect(Booking.prototype.componentDidMount.mock.calls.length).toBe(1)
    });
});
//
// const mockTryGetValue = jest.fn(() => false);
// const mockTrySetValue = jest.fn();
//
// jest.mock('userBooking', () => ({
//     SaveToStorage: jest.fn().mockImplementation(() => ({
//         tryGetValue: mockTryGetValue,
//         trySetValue: mockTrySetValue,
//     })),
// }));
//
// describe('userBooking', () => {
//     it('should set storage on booking button click', () => {
//         mockTryGetValue.mockReturnValueOnce(true);
//         const component = mount(<Booking/>);
//         component.find('button#bookingbutton').simulate('click');
//         expect(mockTryGetValue).toHaveBeenCalled();
//         expect(component).toMatchSnapshot();
//         component.unmount();
//     });
// });