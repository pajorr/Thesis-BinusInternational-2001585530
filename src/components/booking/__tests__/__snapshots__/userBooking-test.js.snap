// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userLogin should render correctly in "debug" mode 1`] = `"Loading..."`;
